src/main/kotlin/no/skatteetaten/aurora/boober/service/DeployService.kt

```diff
@@ -165,12 +163,19 @@ class DeployService(
             val cmd = TagCommand("$dockerGroup/${it.artifactId}", it.version, it.releaseTo!!, dockerRegistry)
             dockerService.tag(cmd)
         }
-        val redeployResponse = redeployService.triggerRedeploy(deploymentSpec, openShiftResponses)
 
-        val redeploySuccess = if (redeployResponse.isEmpty()) true else redeployResponse.last().success
-        val totalSuccess = listOf(success, tagResult?.success, redeploySuccess).filterNotNull().all { it }
+        tagResult?.takeIf { !it.success }?.let {
+            return result.copy(tagResponse = it, reason = "Tag command failed")
+        }
+
```

Comment:
If we here have another step that analyses openShiftStatus to see if we should do either no reploy, a Deployrequest or a IS Import and then maybe a deploy request I think it would be very clean.